ossdbtoolsservice/schema/schema_editor_service.py

```diff
@@ -0,0 +1,186 @@
+
+from ossdbtoolsservice.schema.contracts import (
+        GET_SCHEMA_MODEL_REQUEST,
+        CREATE_SESSION_REQUEST,
+        CLOSE_SESSION_REQUEST,
+        SessionIdContainer,
+)
+
+
+class SchemaEditorService(Service):
+    """Service for browsing database schemas"""
+
+    def __init__(self) -> None:
+        self._service_provider: ServiceProvider | None = None
+        self._conn_service: ConnectionService | None = None
+        self._logger: Logger | None = None
+        self._session_map: dict[str, SchemaEditorSession] = {}
+        self._session_lock: threading.Lock = threading.Lock()
+        self._connect_semaphore = threading.Semaphore(1)
+
+
+    def register(self, service_provider: ServiceProvider) -> None:
+        self._service_provider = service_provider
+        self._logger = self._service_provider.logger
+
+        # Register the request handlers with the server
+        self._service_provider.server.set_request_handler(
+            CREATE_SESSION_REQUEST, self._handle_create_session_request
+        )
+        self._service_provider.server.set_request_handler(
+            GET_SCHEMA_MODEL_REQUEST, self._handle_get_schema_model_request
+        )
+        self._service_provider.server.set_request_handler(
+            CLOSE_SESSION_REQUEST, self._handle_close_session_request
+        )
+        self._service_provider.server.add_shutdown_handler(
+            self._handle_shutdown
+        )
+
+        # Find the provider type
+        self._provider: str = self._service_provider.provider
+
+        if self._service_provider.logger is not None:
+            self._service_provider.logger.info(
+                "Schema Editor service successfully initialized"
+            )
+
+        self._conn_service = self.service_provider.get(
+            constants.CONNECTION_SERVICE_NAME, ConnectionService
+        )
+
+
+    @property
+    def service_provider(self) -> ServiceProvider:
+        if self._service_provider is None:
+            raise ValueError("Service provider is not set")
+        return self._service_provider
+
+
+    def _handle_create_session_request(
+        self, request_context: RequestContext, params: ConnectionDetails
+    ) -> None:
+        """Handle a create object explorer session request"""
+
+        try:
+            self._create_session(request_context, params)
+        except Exception as e:
+            message = f"Failed to create Schema Designer session: {str(e)}"
+            if self.service_provider.logger is not None:
+                self.service_provider.logger.error(message)
+            request_context.send_error(message)
+            return
+
+
+    def _create_session(self, request_context: RequestContext, params: ConnectionDetails) -> None:
+        if self._logger:
+            self._logger.info(f" [handler] Creating OE session for {params.server_name}")
+
+        # Make sure we have the appropriate session params
+        validate.is_not_none("params", params)
```

Comment:
Based on type information, this cannot be None (it would be `ConnectionDetails | None`), so no need to check. A None would have failed in the message deserialization phase earlier than this.